zstacklib/zstacklib/network/ip/link.py

```
@@ -0,0 +1,335 @@
+    
+    with get_iproute(namespace) as ipr:
+        if not indexes and not ifnames:
+            return [IpLink(chunk) for chunk in ipr.get_links()]
+        
+        if ifnames:
+            for ifname in ifnames:
+                indexes.add(check_index_and_ifname(ifname, None, ipr, True))
+        
+        try:
+            return [IpLink(chunk) for chunk in ipr.get_links(*indexes)]
+        except Exception:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)